frontend/theme/index.ts

```diff
@@ -0,0 +1,103 @@
+import { fromEntries } from '../utils'
+
+export const breakpoints = {
+  XS: 600,
+  SM: 786,
+  MD: 1024,
+  LG: 1280,
+} as const
+
+type BreakpointSize = keyof typeof breakpoints
+type Breakpoint = { readonly [key in BreakpointSize]: string }
+
+type Typography =
+  | 'fontFamily'
+  | 'paragraphSM'
+  | 'paragraphMD'
+  | 'paragraphLG'
+  | 'headingOverline'
+  | 'headingSM'
+  | 'headingMD'
+  | 'headingLG'
+  | 'headingXL'
+  | 'displaySM'
+
+export type Theme = {
+  colors: {
+    readonly [key: string]: string
+  }
+  breakpoints: Breakpoint
+  typography: {
+    readonly [key in Typography]: string
+  }
+}
+
+const theme: Theme = {
+  colors: {
+    black: '#393939',
+    white: '#ffffff',
+    gray100: '',
+    gray200: '',
+    gray300: '',
+    gray400: '',
+    gray500: '',
+    gray600: '',
+    gray700: '',
+    gray800: '',
+    blue: '#32567E',
+    blueLight: '#4391C9',
```

Comment:
I really like this naming scheme because it allows for quite a few variations `blueDarkest, blueDarker, blueDark, blue, blueLight, blueLighter, and BlueLightest`. Lemme know what you think!